servo-core/src/main/java/com/netflix/servo/monitor/BucketTimer.java

```diff
@@ -29,9 +29,40 @@
 
 
 /**
- * A simple timer implementation providing the total time, count, min, and max for the times that
- * have been recorded.
+ * A timer implementation using a bucketing approach
+ * that provides a way to get the distribution of samples if the primary
+ * range of values is known.
+ *
+ * For example the following code:
+ *
+ * <pre>
+ BucketTimer t = new BucketTimer(
+     MonitorConfig.builder(name).build(),
+     new BucketConfig.Builder().withBuckets(new long[]{10L, 20L}).build());
+ * </pre>
+ *
+ * will create a <code>BucketTimer</code> that in addition to the statistics:
+ *  <code>totalTime</code>, <code>min</code>, <code>max </code> will report:
+ *  <ul>
+ *     <li>statistic=count bucket=10ms</li>
+ *     <li>statistic=count bucket=20ms</li>
+ *     <li>statistic=count bucket=overflow</li>
+ *  </ul>
+ *
+ *  <ul>
+ *  <li>bucket=10ms will contain the number of samples that were recorded that were
+ *  lower or equal to 10.</li>
+ *
+ *  <li>bucket=20ms will contain the number of samples where the value was lower or equal to 20ms
+ *  and higher than 10.</li>
+ *
+ *  <li>bucket=overflow will contain the remaining entries.</li>
+ * </ul>
+ * Please note that there are no default pre-configured buckets since it is highly dependant
+ * on the use-case. If you fail to specify buckets in {@link BucketConfig} you will get a NPE.
```

Comment:
For spectator we provide some helpers to compute buckets (see BucketFunctions). The main advantage is that we can auto determine the unit and guide users to a more reasonable number. I find users tend to put in a much larger number if left to their own devices and not worry at all about what they actually need. With the unit the problem is that if selected incorrectly it can round multiple buckets to the same label value, e.g., 1500ms getting mapped to 1s if `TimeUnit.SECONDS` is used.